src/DotLiquid/Strainer.cs

```diff
@@ -31,16 +34,26 @@ public static void GlobalFilter(string rawName, object target, MethodInfo method
             FilterFuncs[name] = Tuple.Create(target, methodInfo);
         }
 
+        /// <summary>
+        /// Creates a new instance of the <see cref="Strainer"/> class.
+        /// </summary>
+        /// <param name="context">The DotLiquid context to be used for creating the <see cref="Strainer"/>.</param>
+        /// <returns>A new instance of the <see cref="Strainer"/> class.</returns>
         public static Strainer Create(Context context)
         {
-            Strainer strainer = new Strainer(context);
+            var strainer = new Strainer(context);
+
+            if (context.SyntaxCompatibilityLevel < SyntaxCompatibility.DotLiquidLatest)

```

Comment:
Should this be explicitly `< SyntaxCompatibility.DotLiquid24`?

Reply:
When we introduce DotLiquid25, anything that is 2.4 specific would have been moved to legacy. Unless no change to filters happened (e.g. parse/tags). I'll go your way and comment that it should be adjusted as needed.